src/ir_Daikin.cpp

```diff
@@ -288,60 +727,64 @@ bool IRrecv::decodeDaikin(decode_results *results, uint16_t nbits,
 
   uint32_t data = 0;
   uint16_t offset = OFFSET_START;
-
-  // Header
-  if (!matchMark(results->rawbuf[offset++], DAIKIN_HDR_MARK))
+  unsigned char daikin_code[DAIKIN_COMMAND_LENGTH + 2];
```

Comment:
Question: Why is this + 2 longer than DAIKIN_COMMAND_LENGTH?
Should we make DAIKIN_COMMAND_LENGTH the correct value?

Reply:
This is a (evil) hack so I can pass state data around by reference while decoding (see `DAIKIN_CURBIT` and `DAIKIN_CURINDEX`). So it's not actually required else where.

Reply:
Oh, you are right. That is evil. Too evil in fact, I think. I'll make some suggestions how to do this using readData(). That said, I like the bounds checking, which readData() doesn't have. Hmmm